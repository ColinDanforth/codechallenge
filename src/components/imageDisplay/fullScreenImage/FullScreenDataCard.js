import React from 'react'
import PropTypes from 'prop-types'
import close from '../../localImges/close.png'
import FullScreenComments from "./FullScreenComments"
import FullScreenUserCard from "./FullScreenUserCard"

const rootDiv={
  position: 'absolute',
  top: '0',
  left: '0',
  marginTop: '100vh',
  maxHeight: '300px',
  minWidth: '100vw',
  width: '100wh',
  height: 'auto',
  backgroundColor: 'RGBA(256,256,256, 0.6)',
  overflowY: 'scroll',
  zIndex: '31',
  display: 'flex',
  flexDirection: 'row',
  justifyContent: 'flex-start',
}

const xStyle = {
  maxWidth: '45px',
  minWidth: '35px',
  width: 'auto',
  height: 'auto',
  cursor: 'pointer',
}

const commentsBlock={
  flex: '1',
}

const closeWrapperDiv = {
  position: 'fixed',
  top: '0',
  left: '0',
  backgroundColor: 'RGBA(256,256,256,1)',
  height: 'auto',
  width: 'auto',
  borderRadius: '20%',
}

class FullScreenDataCard extends React.Component{
  constructor(){
    super()

    this.handleCloseFullScreen = this.handleCloseFullScreen.bind(this)
  }

  handleCloseFullScreen(){
    this.props.closeFullScreen()
  }

  render(){
    console.log(JSON.stringify(this.props.thisImagesComments))
    return(
      <div style={rootDiv}>
        <div style={closeWrapperDiv}>
          <img src={close} alt="close" style={xStyle} onClick={this.handleCloseFullScreen}/>
        </div>
        <FullScreenUserCard thisItem={this.props.thisItem} dataObject={this.props.thisDataObject}/>
        <div style={commentsBlock}>
          <FullScreenComments thisImagesComments={this.props.thisImagesComments}/>
        </div>
      </div>
    )
  }
}

export default FullScreenDataCard

FullScreenDataCard.propTypes= {
  thisItemId: PropTypes.number,
  thisItem: PropTypes.object,
  thisDataObject: PropTypes.object,
  thisImagesComments: PropTypes.object,
  closeFullScreen: PropTypes.func,
}
