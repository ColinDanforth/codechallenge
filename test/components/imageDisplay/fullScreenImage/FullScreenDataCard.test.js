import React from 'react';
import FullScreenDataCard from '../../../../src/components/imageDisplay/fullScreenImage/FullScreenDataCard';
import renderer from 'react-test-renderer';

test('FullScreenUserCard should display the below data object in FullScreenUserCard, FulScreenComments and Title elements',
  () => {
    const testDataObject={
      name: 'test name',
      user: {
        avatars: {
          default: {
            https: 'http://google.ca'
          }
        },
        fullname: 'Test User',

      },
      rating: '99.99999995',
      lens_info: {
        friendly_name: "Olympus M.Zuiko Digital ED 12-40mm f/2.8 PRO",
        features: "Micro Four Thirds System<br/>24-80mm (35mm Equivalent)<br/>Aperture Range: f/2.8 to f/22",
      },
      camera: "Olympus M.Zuiko Digital ED 12-40mm f/2.8 PRO",
      shutter_speed: 'test string',
      aperture: 'test string',
      description: 'test description',
      comments: [{
        user: {
          username: 'TestUser'
        },
        created_at: '2018-01-01',
        body: 'Some Comment String',
      },{
        user: {
          username: 'TestUser2'
        },
        created_at: '2018-01-02',
        body: 'Some Comment String 2',
      }],
    }

    const component = renderer.create(
      <FullScreenDataCard thisDataObject={testDataObject} thisImagesComments={testDataObject} thisItem={testDataObject}/>
    )

    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()

  })
