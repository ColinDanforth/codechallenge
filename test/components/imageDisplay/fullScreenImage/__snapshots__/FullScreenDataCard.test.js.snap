// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FullScreenUserCard should display a group of comment elements showing the appropriate text from the testData object 1`] = `
<div
  style={
    Object {
      "backgroundColor": "RGBA(0,0,0,0.5)",
      "display": "flex",
      "flexDirection": "row",
      "height": "auto",
      "justifyContent": "flex-start",
      "left": "0",
      "marginTop": "100vh",
      "maxHeight": "300px",
      "minWidth": "100vw",
      "overflowY": "scroll",
      "position": "absolute",
      "top": "0",
      "width": "100wh",
      "zIndex": "31",
    }
  }
>
  <div
    style={
      Object {
        "backgroundColor": "RGB(256,256,256)",
        "borderRadius": "20%",
        "height": "auto",
        "left": "0",
        "position": "fixed",
        "top": "0",
        "width": "auto",
      }
    }
  >
    <img
      alt="close"
      onClick={[Function]}
      src="test-file-stub"
      style={
        Object {
          "cursor": "pointer",
          "height": "auto",
          "maxWidth": "45px",
          "minWidth": "35px",
          "width": "auto",
        }
      }
    />
  </div>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "height": "auto",
        "justifyContent": "flex-start",
        "marginLeft": "2%",
        "marginRight": "2%",
        "marginTop": "1%",
      }
    }
  >
    <div
      style={
        Object {
          "marginBottom": "2%",
          "maxWidth": "100vw",
          "width": "auto",
        }
      }
    >
      <div
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFF",
            "borderBottom": "inset",
            "display": "flex",
            "flexDirection": "Column",
            "height": "auto",
            "justifyContent": "center",
            "minWidth": "600px",
            "width": "auto",
          }
        }
      >
        <h1>
          test name
        </h1>
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "test description",
            }
          }
          style={
            Object {
              "height": "auto",
              "margin": "3%",
              "maxWidth": "820px",
              "minWidth": "600px",
              "width": "auto",
            }
          }
        />
      </div>
    </div>
    <div
      style={
        Object {
          "height": "auto",
          "maxWidth": "820px",
          "minWidth": "820px",
          "width": "auto",
        }
      }
    >
      <div
        style={
          Object {
            "backgroundColor": "#FFF",
            "borderBottom": "inset",
            "borderRight": "inset",
            "display": "flex",
            "flexDirection": "row",
            "height": "auto",
            "justifyContent": "flex-start",
            "maxWidth": "820px",
            "minWidth": "820px",
          }
        }
      >
        <img
          alt="avatar"
          src="http://google.ca"
          style={
            Object {
              "height": "auto",
              "margin": "0.3%",
              "maxHeight": "100px",
              "width": "auto",
            }
          }
        />
        <div
          style={
            Object {
              "display": "flex",
              "flexDirection": "row",
              "height": "auto",
              "justifyContent": "space-between",
              "marginLeft": "10px",
              "maxWidth": "700px",
              "minWidth": "700px",
              "width": "auto",
            }
          }
        >
          <div>
            <p
              style={
                Object {
                  "height": "auto",
                  "maxWidth": "400px",
                  "minWidth": "400px",
                  "width": "auto",
                }
              }
            >
              Photographer: 
              Test User
            </p>
            <p
              style={
                Object {
                  "height": "auto",
                  "maxWidth": "400px",
                  "minWidth": "400px",
                  "width": "auto",
                }
              }
            >
              Rating: 
              99.99999995
              %
            </p>
          </div>
          <div>
            <img
              alt="like"
              src="test-file-stub"
              style={
                Object {
                  "height": "auto",
                  "marginRight": "10px",
                  "marginTop": "5px",
                  "maxWidth": "40px",
                  "width": "auto",
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "height": "auto",
          "marginBottom": "4%",
          "maxWidth": "820px",
          "minWidth": "820px",
          "width": "auto",
        }
      }
    >
      <div
        style={
          Object {
            "backgroundColor": "RGBA(256,256,256,1)",
            "borderBottom": "inset",
            "marginTop": "2%",
          }
        }
      >
        <p
          style={
            Object {
              "marginBottom": "15px",
              "marginLeft": "30px",
              "marginRight": "30px",
            }
          }
        >
          Camera: 
          Olympus M.Zuiko Digital ED 12-40mm f/2.8 PRO
        </p>
        <p
          style={
            Object {
              "marginBottom": "15px",
              "marginLeft": "30px",
              "marginRight": "30px",
            }
          }
        >
          Shutter Speed: 
          test string
        </p>
        <p
          style={
            Object {
              "marginBottom": "15px",
              "marginLeft": "30px",
              "marginRight": "30px",
            }
          }
        >
          Aperture: 
          test string
        </p>
        <div>
          <p
            style={
              Object {
                "marginBottom": "15px",
                "marginLeft": "30px",
                "marginRight": "30px",
              }
            }
          >
            Lens Info:
          </p>
          <p
            style={
              Object {
                "marginBottom": "15px",
                "marginLeft": "60px",
                "marginRight": "30px",
              }
            }
          >
            Name: 
            Olympus M.Zuiko Digital ED 12-40mm f/2.8 PRO
          </p>
          <p
            style={
              Object {
                "marginBottom": "15px",
                "marginLeft": "60px",
                "marginRight": "30px",
              }
            }
          >
            Feature: 
            Micro Four Thirds System
          </p>
          <p
            style={
              Object {
                "marginBottom": "15px",
                "marginLeft": "60px",
                "marginRight": "30px",
              }
            }
          >
            24-80mm (35mm Equivalent)
          </p>
          <p
            style={
              Object {
                "marginBottom": "15px",
                "marginLeft": "60px",
                "marginRight": "30px",
              }
            }
          >
            Aperture Range: f/2.8 to f/22
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "flex": "1",
      }
    }
  >
    <div
      style={
        Object {
          "backgroundColor": "RGB(256,256,256)",
          "borderBottom": "inset",
          "display": "flex",
          "flexDirection": "column",
          "height": "auto",
          "justifyContent": "center",
          "marginBottom": "3%",
          "marginTop": "3%",
          "maxWidth": "500px",
          "width": "auto",
        }
      }
    >
      <p
        style={
          Object {
            "alignSelf": "center",
            "fontSize": "1.3em",
            "fontWeight": "bold",
          }
        }
      >
        User Comments
      </p>
      <div
        style={
          Object {
            "background": "RGBA(256,256,256,1)",
            "border": "outset",
            "display": "flex",
            "flexDirection": "column",
            "justifyContent": "center",
            "margin": "0.5%",
          }
        }
      >
        <div
          style={
            Object {
              "alignItems": "flex-start",
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginLeft": "1%",
              "marginRight": "1%",
            }
          }
        >
          <p
            style={
              Object {
                "fontWeight": "bold",
                "maxWidth": "500px",
                "width": "auto",
                "wordWrap": "normal",
              }
            }
          >
            TestUser
          </p>
          <p>
            2018-01-01
          </p>
        </div>
        <div
          style={
            Object {
              "marginLeft": "2%",
              "marginRight": "2%",
              "width": "auto",
            }
          }
        >
          <p
            style={
              Object {
                "maxWidth": "500px",
                "width": "auto",
                "wordWrap": "normal",
              }
            }
          >
            Some Comment String
          </p>
        </div>
      </div>
      <div
        style={
          Object {
            "background": "RGBA(256,256,256,1)",
            "border": "outset",
            "display": "flex",
            "flexDirection": "column",
            "justifyContent": "center",
            "margin": "0.5%",
          }
        }
      >
        <div
          style={
            Object {
              "alignItems": "flex-start",
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginLeft": "1%",
              "marginRight": "1%",
            }
          }
        >
          <p
            style={
              Object {
                "fontWeight": "bold",
                "maxWidth": "500px",
                "width": "auto",
                "wordWrap": "normal",
              }
            }
          >
            TestUser2
          </p>
          <p>
            2018-01-02
          </p>
        </div>
        <div
          style={
            Object {
              "marginLeft": "2%",
              "marginRight": "2%",
              "width": "auto",
            }
          }
        >
          <p
            style={
              Object {
                "maxWidth": "500px",
                "width": "auto",
                "wordWrap": "normal",
              }
            }
          >
            Some Comment String 2
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
